scripts/metadata/metadata.py

```diff
@@ -260,24 +297,49 @@ def __init__(self, matching_filepaths, output_folder : Path):
             
             print("Done")
             print("="*80)
-        
-    def _check_entry_mismatch(self, df, columns):
-        """
-        Check that data is in agreement for supplied columns 
-        """
-        
-        for c in columns:
-            #Identify mismatches and show the errors
-            mismatches_df = df.loc[(df[f"{c}_x"] != df[f"{c}_y"]) ]
-            cols_to_retain = ['expt_id_x','expt_id_y', f"{c}_x", f"{c}_y"]
-            if mismatches_df.shape[0] > 0:
-                print(f"WARNING: Mismatches identified for {c}")
-                print(mismatches_df[cols_to_retain])
+
 
     def _count_non_none_entries_in_dfcolumn (self, df, column):
         '''
         Function counts the number of non none entries in a column of a dataframe
         '''
         
         return len([item for item in list(chain.from_iterable(df[f"{column}"])) if not item=="None"])
+    
+    def _report_entry_mismatches(self, df, columns):
+        """
+        Summarisesd and reports all of the files and entries with mismatching data 
+        """
+        if mismatches_df.shape[0] > 0:
+            print(f"WARNING: Mismatches identified for {c}")
+            print(mismatches_df[cols_to_retain])
+    
+    def _check_duplicate_entries(self, dictionary : dict, attribute : str) -> list :
```

Comment:
Are you sure the try / except is necessary in this function body? From my reading of the code it doesn't look to me like it should ever get called. Maybe this works:
```
value_counts = {}
for _,  object in dt.items():
	value = getattr(object, attribute)
	if  not  value:
		continue
	if value in value_counts:
		value_counts[value] += 1
	else:
		value_counts[value] = 1
return [k for k,v in value_counts if v > 1]
```

Reply:
Thanks Jason - great suggestions - have tidied up the code and think it is much better now....